components/application-mgt/org.wso2.carbon.identity.application.common/src/main/java/org/wso2/carbon/identity/application/common/dao/impl/AuthenticatorManagementDAOImpl.java

```diff
@@ -0,0 +1,318 @@
+                    config.setName(rs.getString(Column.NAME));
+                    config.setDisplayName(rs.getString(Column.DISPLAY_NAME));
+                    config.setEnabled(rs.getString(Column.IS_ENABLED).equals(IS_TRUE_VALUE));
+                    config.setDefinedByType(DefinedByType.valueOf(rs.getString(Column.DEFINED_BY)));
+                }
+            }
+
+            if (config == null) {
+                return null;
+            }
+
+            int authenticatorConfigID = getAuthenticatorIdentifier(dbConnection, config.getName(), tenantId);
+            try (NamedPreparedStatement statementProp = new NamedPreparedStatement(dbConnection,
+                    Query.GET_AUTHENTICATOR_PROP_SQL)) {
+                statementProp.setInt(Column.AUTHENTICATOR_ID, authenticatorConfigID);
+                statementProp.setInt(Column.TENANT_ID, tenantId);
+
+                try (ResultSet rs = statementProp.executeQuery()) {
+                    List<Property> properties = new ArrayList<>();
+                    while (rs.next()) {
+                        Property property = new Property();
+                        property.setName(rs.getString(Column.PROPERTY_KEY));
+                        property.setValue(rs.getString(Column.PROPERTY_VALUE));
+                        property.setConfidential(Boolean.parseBoolean(rs.getString(Column.IS_SECRET)));
+                        properties.add(property);
+                    }
+                    config.setProperties(properties.toArray(new Property[0]));
+                }
+            }
+
+            IdentityDatabaseUtil.commitTransaction(dbConnection);
+            return config;
+        } catch (SQLException e) {
+            throw buildServerException(AuthenticatorMgtError.ERROR_WHILE_RETRIEVING_AUTHENTICATOR_BY_NAME, e);
+        }
+    }
+
+    private UserDefinedLocalAuthenticatorConfig getLocalAuthenticatorConfigBasedOnType(String authenticationType) {
+
+        if (AuthenticationType.VERIFICATION.toString().equals(authenticationType)) {
+            return new UserDefinedLocalAuthenticatorConfig(AuthenticationType.VERIFICATION);
+        }
+        return new UserDefinedLocalAuthenticatorConfig(AuthenticationType.IDENTIFICATION);
+    }
+
+    private boolean isBasicInfoUpdated(UserDefinedLocalAuthenticatorConfig existingAuthenticatorConfig,
+                                       UserDefinedLocalAuthenticatorConfig updatedAuthenticatorConfig) {
+
+        return !existingAuthenticatorConfig.getDisplayName().equals(updatedAuthenticatorConfig.getDisplayName()) ||
+                existingAuthenticatorConfig.isEnabled() != updatedAuthenticatorConfig.isEnabled();
+    }
+
+    private int getAuthenticatorIdentifier(Connection dbConnection, String authenticatorConfigName,
+                                               int tenantId) throws AuthenticatorMgtServerException, SQLException {
```

Comment:
Why do you build and throw this AuthenticatorMgtServer Exception here ?
IMO, this should just return the SQL Exception and should have been handled at the same transaction